
* probably not kfreebsd specific bug, 
  but I am not sure as ld.so loading is a little magic


--- sysdeps/x86_64/dl-machine.h	2005-07-31 19:49:44.000000000 +0200
+++ sysdeps/x86_64/dl-machine.h	2007-02-08 18:14:56.000000000 +0100
@@ -286,7 +283,7 @@
       const Elf64_Sym *const refsym = sym;
 #endif
       struct link_map *sym_map = RESOLVE_MAP (&sym, version, r_type);
-      Elf64_Addr value = (sym == NULL ? 0
+      Elf64_Addr value = (sym_map == NULL ? 0
 			  : (Elf64_Addr) sym_map->l_addr + sym->st_value);
 
 #if defined RTLD_BOOTSTRAP && !USE___THREAD
